### ISSUE TITLE
Cast TransitioningContentControl to ContentControl removes transitioning

### ISSUE THREAD

**Issue by dojo90** [AvaloniaUI/Avalonia#6759]
I don't know if this is a bug, but if I cast the `TransitioningContentControl` to `ContentControl`, it removes the transitioning effect.

```cs
        private TransitioningContentControl _transitioningContentControl;
        private ContentControl _contentControl;

        public void SetContentPresenter(TransitioningContentControl contentControlFromUi)
        {
            _transitioningContentControl = contentControlFromUi;
            _contentControl = contentControlFromUi;
        }

        public void Navigate(Type myFooViewType)
        {
            // works
            _transitioningContentControl.Content = App.Container.GetRequiredService(myFooViewType);
            // doesn't work
            _contentControl.Content = App.Container.GetRequiredService(myFooViewType);
        }
```

**Avalonia.ReactiveUI v0.10.8**

**Comment by maxkatz6** [comment]
Because TransitioningContentControl uses "new object Content {get;set;}", which doesn't work after casting.
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.ReactiveUI/TransitioningContentControl.cs#L51

There should be another way to play transition before and after setting new value.

**Comment by maxkatz6** [comment]
It was fixed in 11.0
